lib/ABI/AddABIAwarenessToFunctionPrototypes.cpp

```diff
@@ -0,0 +1,210 @@
+                                       const model::TypePath &Type) {
+  return !needsVectorRegisterSupportImpl(VH, model::QualifiedType(Type, {}));
+}
+
+using namespace std::string_literals;
+
+class AddABIAwarenessToFunctionPrototypes {
+public:
+  static constexpr auto Name = "AddABIAwarenessToFunctionPrototypes";
+  inline static const std::tuple Options = { pipeline::Option("abi",
+                                                              "Invalid"s) };
+  std::vector<std::vector<pipeline::Kind *>> AcceptedKinds = {};
+
+  void run(pipeline::Context &Context, std::string InputABI) {
+    auto &Model = revng::getWritableModelFromContext(Context);
+
+    // Minimize the negative impact on binaries with ABI that is not fully
+    // supported by disabling the conversion by default.
+    //
+    // Use revng model pass on those functions to convert them manually.
+    //
+    // TODO: extend this list.
+    static constexpr std::array ABIsTheConversionIsEnabledFor = {
+      model::ABI::SystemV_x86_64,        model::ABI::SystemV_x86,
+      model::ABI::SystemV_x86_regparm_3, model::ABI::SystemV_x86_regparm_2,
+      model::ABI::SystemV_x86_regparm_1, model::ABI::AAPCS
+    };
+    if (!llvm::is_contained(ABIsTheConversionIsEnabledFor, Model->DefaultABI()))
+      return;
+
+    // This reuses the verification map within the `model::VerifyHelper` in
+    // an incompatible manner. DO NOT pass this object into a normal
+    // verification routine or things are going to break down.
+    model::VerifyHelper VectorVH;
+
+    // This is a normal `model::VerifyHelper`.
+    model::VerifyHelper ValidationVH;
+
+    std::optional<model::ABI::Values> ABI = model::ABI::fromName(InputABI);
+    if (ABI.value() == model::ABI::Values::Invalid)
+      ABI = std::nullopt;
+
+    // Choose the applicable functions and run the conversion with the default
+    // ABI (`Model->DefaultABI`) for them.
+    using abi::FunctionType::chooseTypes;
+    auto ToConvert = chooseTypes<model::RawFunctionType>(Model->Types());
+    for (model::RawFunctionType *Old : ToConvert) {
+      if (needsVectorRegisterSupport(VectorVH, Model->getTypePath(Old))) {
+        // Skip functions requiring vector register support.
+        // TODO: remove this check after `abi::FunctionType` supports vectors.
+        continue;
+      }
+
+      if (auto New = abi::FunctionType::tryConvertToCABI(*Old, Model, ABI)) {
+        // If the conversion succeeds, make sure the returned type is valid,
+        revng_assert(New->isValid());
+
+        // and verifies.
+        revng_assert(New->get()->verify(ValidationVH));
+      } else {
+        // Do nothing if the conversion failed (the model is not modified).
+        // `RawFunctionType` is still used for those functions.
+        // This might be an indication of an ABI misdetection.
+      }
+    }
+  }
+};
+
+pipeline::RegisterAnalysis<AddABIAwarenessToFunctionPrototypes> _;
```

Comment:
`_` is not a valid name I think. Use something like `RA1`.

Reply:
Done.